### PR TITLE
feat(dataarts): add a new resource to manage factory job

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccFactoryJob_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts -v

=== RUN   TestAccFactoryJob_basic
=== PAUSE TestAccFactoryJob_basic
=== CONT  TestAccFactoryJob_basic
--- PASS: TestAccFactoryJob_basic (15.10s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts	15.147s

```

### CODE REVIEW

File: docs/resources/dataarts_factory_job.md [hunk 1]
```diff
@@ -0,0 +1,455 @@
+---
+subcategory: "DataArts Studio"
+---
+
+# huaweicloud_dataarts_factory_job
+
+Manages a job resource of DataArts within HuaweiCloud.
```
Comment (ShiChangkuo) [inline docs/resources/dataarts_factory_job.md]:
```suggestion
Manages a job resource of DataArts Factory within HuaweiCloud.
```

File: docs/resources/dataarts_factory_job.md [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+* `workspace_id` - (Optional, String, ForceNew) The workspace ID.
+  If this parameter is not set, the default workspace is used by default.
+  Changing this parameter will create a new resource.
+
+* `params` - (Optional, List) Job parameter definition.
+  The [params](#job_Param) structure is documented below.
+
+* `directory` - (Optional, String) Path of a job in the directory tree.  
+  If the directory of the path does not exist during job creation, a directory is automatically
+   created in the root directory /, for example, /dir/a/.
```
Comment (ShiChangkuo) [inline docs/resources/dataarts_factory_job.md]:
```suggestion
  created in the root directory /, for example, /dir/a/.
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_factory_job.go [hunk 3]
```diff
@@ -0,0 +1,1328 @@
...
+	}
+
+	if v, ok := d.GetOk("workspace_id"); ok {
+		getJobOpt.MoreHeaders["workspace"] = v.(string)
+	}
+
+	getJobResp, err := getJobClient.Request("GET", getJobPath, &getJobOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Job")
```
Comment (ShiChangkuo) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_factory_job.go]: it seems that we should convert the error code when job is not exist.
The status is 400, and the response is as following:
```json
{
	"error_code":"DLF.0100",
	"error_msg":"Job[nnnnn] does not exist"
}
```

Comment (hmbig2) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_factory_job.go]: added

File: docs/resources/dataarts_factory_job.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/dataarts_factory_job.md]: done

File: docs/resources/dataarts_factory_job.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/dataarts_factory_job.md]: done

**Pull request merged**